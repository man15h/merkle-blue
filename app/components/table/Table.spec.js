import { mount, shallowMount } from '@vue/test-utils';
import { render } from '@vue/server-test-utils';
import MbTable from './Table';

describe('MbTable', () => {
  it('is called', () => {
    const wrapper = mount(MbTable);
    expect(wrapper.name()).toBe('MbTable');
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
  it('renders table with 2 data columns', () => {
    const wrapper = shallowMount(MbTable, {
      propsData: propsData
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});

const propsData = {
  data: [
    {
      Amount: 345.54,
      Date: '2017-07-23T04:24:49-07:00',
      Description: 'Curabitur dictum. Phasellus in',
      ID: '3471DA17-401F-9633-BF81-4CADA6FD5C79',
      Name: 'Kyra Lester'
    },
    {
      Amount: 677.08,
      Date: '2018-11-08T05:44:15-08:00',
      Description: 'dui, in sodales elit erat vitae risus. Duis a mi',
      ID: '9F5C9912-936A-FB85-1EDB-9DA87BE7FF1E',
      Name: 'Buckminster Alvarado'
    }
  ],
  columns: [
    {
      field: 'ID',
      label: 'ID'
    },
    {
      field: 'Name',
      label: 'Name'
    }
  ]
};
