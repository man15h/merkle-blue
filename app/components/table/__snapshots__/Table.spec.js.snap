// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BcTable renders table with 2 data columns 1`] = `
<div>
  <table class="text-left w-full bc-table" style="border-collapse: collapse;">
    <thead>
      <tr>
        <!---->
        <th class="table-header">
          <div>
            ID
            <!---->
          </div>
        </th>
        <th class="table-header">
          <div>
            Name
            <!---->
          </div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <!---->
        <bctablecolumn-stub label="ID" field="ID" width="" visible="true" internal="true">
          3471DA17-401F-9633-BF81-4CADA6FD5C79
        </bctablecolumn-stub>
        <bctablecolumn-stub label="Name" field="Name" width="" visible="true" internal="true">
          Kyra Lester
        </bctablecolumn-stub>
      </tr>
      <tr>
        <!---->
        <bctablecolumn-stub label="ID" field="ID" width="" visible="true" internal="true">
          9F5C9912-936A-FB85-1EDB-9DA87BE7FF1E
        </bctablecolumn-stub>
        <bctablecolumn-stub label="Name" field="Name" width="" visible="true" internal="true">
          Buckminster Alvarado
        </bctablecolumn-stub>
      </tr>
    </tbody>
  </table>
  <!---->
</div>
`;
